index.js

```diff
@@ -17,12 +17,25 @@ function shouldComponentUpdate(nextProps, nextState) {
   return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState);
 }
 
+function getFunctionName(fun) {
+  var ret = fun.toString();
+  ret = ret.substr('function '.length);
+  ret = ret.substr(0, ret.indexOf('('));
+  return ret;
+}
+
 /**
  * Makes the given component "pure".
  *
  * @param object component Component.
  */
 function pureRenderDecorator(component) {
+  if (component.prototype.shouldComponentUpdate !== undefined) {
+    var componentName = component.prototype.name || component.name || getFunctionName(component);
```

Comment:
Do you think we could use [`displayName`](https://facebook.github.io/react/docs/react-component.html#displayname) here instead of guessing the name ? I also saw a [`getName`](https://github.com/facebook/react/search?p=1&q=getName&utf8=%E2%9C%93) function used here and there in React but I didn't go much further.